### PR TITLE
move useGetAllBalance and useGetCoinBalance to core

### PR DESCRIPTION
**Pull request by Jibz-Mysten**
## Description 

- Move the useGetAllBalance and useGetCoinBalance to core
- Fix the reordering issue on the Token

### CODE REVIEW

File: apps/core/src/hooks/useGetAllBalances.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import { useQuery } from '@tanstack/react-query';
+
+export function useGetAllBalances(
+    address?: SuiAddress | null,
+    refetchInterval?: number,
+    staleTime?: number
+) {
+    const rpc = useRpcClient();
+    return useQuery({
+        queryKey: ['get-all-balance', address, staleTime, refetchInterval],
```
Comment (pchrysochoidis) [inline apps/core/src/hooks/useGetAllBalances.ts:15]: Is it necessary to add `staleTime` and `refetchInterval` in the query key? Does it affect the data?

File: apps/wallet/src/ui/app/hooks/useGetAllBalances.ts [hunk 2]
```diff
@@ -8,6 +8,7 @@ import { useQuery } from '@tanstack/react-query';
 
 import { FEATURES } from '_src/shared/experimentation/features';
 
+// TODO: use from core. Update core to handle optional select function
```
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/hooks/useGetAllBalances.ts:11]: Would this be a follow up PR or missed it?

File: apps/core/src/hooks/useGetCoinBalance.ts [hunk 3]
```diff
@@ -1,28 +1,29 @@
...
-        20_000
-    );
 
     return useQuery({
-        queryKey: ['coin-balance', address, coinType],
+        queryKey: [
+            'coin-balance',
+            address,
+            coinType,
+            staleTime,
```
Comment (pchrysochoidis) [inline apps/core/src/hooks/useGetCoinBalance.ts:21]: same here about staleTime and refetchInterval

File: apps/wallet/src/ui/app/helpers/sortGetAllBalancesToken.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { Coin, type CoinBalance } from '@mysten/sui.js';
+
+// Sort tokens by symbol and total balance
+// Move this to the API backend
+export function sortGetAllBalancesToken(tokens: CoinBalance[]) {
```
Comment (Jibz-Mysten) [inline apps/wallet/src/ui/app/helpers/sortGetAllBalancesToken.ts:8]: This address the reordering issue. Ideally the API should sort the tokens

File: apps/wallet/src/ui/app/hooks/useGetAllBalances.ts (hunk 2 above)
Comment (Jibz-Mysten) [reply apps/wallet/src/ui/app/hooks/useGetAllBalances.ts]: Updated this PR to resolve this

**Review by pchrysochoidis — APPROVED** [review]

**Pull request merged**